### DISCUSSION TITLE
Battery compatibility

### DISCUSSION THREAD

**Discussion by alexswerner** [iRobotEducation/create3_docs#608] [Questions]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Which version of ROS 2 is installed on your computer?

None

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

FastRTPS

### Are there multiple Create 3 robots connected to your network?

None

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

Hi,

I'm wondering if I can get certainty about the battery compatibility of the Create3. @shamlian stated in Question #383 that the Roomba Combo™ j7+ should be compatible and have twice the capacity. This would be very helpful in our application. Can I get a confirmation that (a) this battery is compatible and (b) has actually twice the capacity.

Thanks!

**Comment by shamlian** [comment]
I can confirm that [this battery](https://www.irobot.com/en_US/lithium-battery-for-roomba-combo-j9plus-and-j7plus%252c-and-roomba-j9plus/4785636.html) is rated to 64.2 Wh and when plugged into a Create 3 robot shows, on the info page,
```
>>>>>>> system battery
Percentage:24
Capacity(mAh):4740
```
It appears to be charging as normal.

The battery I previously had in my robot (originally as shipped) showed
```
Capacity(mAh):1927
```

This is roughly 2.5x the capacity of the original battery.

The robot by itself should fully charge as expected. However, depending on the hotel load of other things plugged into the system, the battery may not charge fast enough to satisfy the robot's safety system, and the robot may stop charging before it is fully charged (this is a safety feature). In this case, simply backing the robot off and then back onto the dock should get it to start charging again. Current into or out of the battery is available in the `/battery_state` topic.

I will update this if I have any trouble.
